src/common/checksum.c

```diff
@@ -588,51 +588,168 @@ daos_csummer_alloc_iods_csums(struct daos_csummer *obj, daos_iod_t *iods,
 }
 
 static int
-calc_csum_recx(struct daos_csummer *obj, d_sg_list_t *sgl,
-	       size_t rec_len, daos_recx_t *recxs, size_t nr,
-	       struct dcs_csum_info *csums)
+calc_csum_recx_with_no_map(struct daos_csummer *obj, size_t csum_nr,
+			   daos_recx_t *recx,
+			   struct dcs_csum_info *csum_info,
+			   size_t rec_len, d_sg_list_t *sgl,
+			   uint32_t rec_chunksize,
+			   struct daos_sgl_idx *idx)
 {
+	uint32_t		 i;
```

Comment:
personal nit...order by decreasing size, similar to what you would do for a struct